context/src/junit/bindings.rs

```diff
@@ -655,3 +663,71 @@ impl Into<NonSuccessKind> for BindingsNonSuccessKind {
         }
     }
 }
+
+#[cfg_attr(feature = "pyo3", gen_stub_pyclass, pyclass(get_all))]
+#[cfg_attr(feature = "wasm", wasm_bindgen(getter_with_clone))]
+#[derive(Clone, Debug)]
+pub struct BindingsJunitReportValidation {
+    all_issues: Vec<JunitReportValidationFlatIssue>,
+    level: JunitValidationLevel,
+    test_suites: Vec<JunitTestSuiteValidation>,
+    valid_test_suites: Vec<BindingsTestSuite>,
+}
+
+impl From<JunitReportValidation> for BindingsJunitReportValidation {
+    fn from(
+        JunitReportValidation {
+            all_issues,
+            level,
+            test_suites,
+            valid_test_suites,
+        }: JunitReportValidation,
+    ) -> Self {
+        Self {
+            all_issues: all_issues
+                .into_iter()
+                .map(|i| JunitReportValidationFlatIssue {
+                    level: JunitValidationLevel::from(&i),
+                    error_type: JunitValidationType::from(&i),
+                    error_message: i.to_string(),
+                })
+                .collect(),
+            level,
+            test_suites,
+            valid_test_suites: valid_test_suites
+                .into_iter()
+                .map(BindingsTestSuite::from)
+                .collect(),
+        }
+    }
+}
+
+#[cfg_attr(feature = "pyo3", gen_stub_pymethods, pymethods)]
+#[cfg_attr(feature = "wasm", wasm_bindgen)]
+impl BindingsJunitReportValidation {
+    pub fn all_issues_owned(&self) -> Vec<JunitReportValidationFlatIssue> {
+        self.all_issues.clone()
+    }
+
+    pub fn max_level(&self) -> JunitValidationLevel {
+        self.test_suites
+            .iter()
+            .map(|test_suite| test_suite.max_level())
+            .max()
+            .map_or(self.level, |l| l.max(self.level))
+    }
+
+    pub fn num_invalid_issues(&self) -> usize {
+        self.all_issues
+            .iter()
+            .filter(|issue| issue.level == JunitValidationLevel::Invalid)
+            .count()
+    }
+
+    pub fn num_suboptimal_issues(&self) -> usize {
+        self.all_issues
+            .iter()
+            .filter(|issue| issue.level == JunitValidationLevel::SubOptimal)
+            .count()
+    }
+}
```

Comment:
This seems to be related to the comment from the PR description:
> Introduces a new BindingsJunitReportValidation type. Since JunitReportValidation now includes a vector of quick_junit::TestSuite as the 'valid' test suites, pyo3 can no longer directly create a class from it

I'm not sure that's true? That's only if you are implementing `pyclass(get_all)`, not `pyclass`. Do we need to implement `pyclass(get_all)` or can we just implement `pymethods` for the things that Python needs access to?

Reply:
The root of the issue is that I'd like to return the actual `quick_junit::TestSuite` structs as the 'valid' test suites, and I don't see a way to do that without first converting them to our `BindingsTestSuite` structs first. Is there a way to do that conversion as a `pymethod` on `JunitReportValidation`?

Reply:
I think I see your point.

The nicest way we can try to do that is by newtype wrapping `quick_junit::TestSuite` and having it `Deref` to its inner `quick_junit::TestSuite`. That way we can use Python methods on it and also use it within our Rust code. I now that would take more work, but it's a pattern that is more scalable, so we should consider if we find ourselves needing more of this same pattern.

Another possibility would be to have multiple `impl`s with some methods specifcally for Python to use. An issue with that, is that currently `pymethod` cannot be used multiple times, so you could not use some of the Rust `impl` with the Python `impl`.